build-system/tasks/integration.js

```diff
@@ -42,25 +53,38 @@ class Runner extends RuntimeTestRunner {
 }
 
 async function buildTransformedHtml() {
-  const filePaths = await globby('./test/fixtures/**/*.html');
-  let normalizedFilePath;
+  const filePaths = await globby(INTEGRATION_FIXTURES);
+  fs.ensureDirSync('./test-bin/');
+  for (const filePath of filePaths) {
+    const normalizedFilePath = pathModule.normalize(filePath);
+    await transformAndWriteToTestFolder(normalizedFilePath);
+  }
+}
+
+async function transformAndWriteToTestFolder(filePath) {
   try {
-    log(
-      green('Copying integration test files to'),
-      cyan('test-bin/') + green('...')
-    );
-    for (const filePath of filePaths) {
-      await fs.copySync(filePath, `./test-bin/${filePath}`);
-    }
+    const html = await htmlTransform(filePath);
+    const fullFilePath = `./test-bin/${filePath}`;
+    const targetDir = pathModule.dirname(fullFilePath);
+    fs.ensureDirSync(targetDir);
+    fs.writeFileSync(fullFilePath, html);
   } catch (e) {
-    console./*OK*/ log(
-      `${normalizedFilePath} could not be transformed by the postHTML ` +
-        `pipeline.\n${e.message}`
+    log(
+      yellow('WARNING:'),
+      cyan(
+        `${filePath} could not be transformed by the postHTML ` +
+          'pipeline. Falling back to copying.'
+      ),
+      yellow(`Reason: ${e.message}`)
     );
+    fs.copySync(filePath, `./test-bin/${filePath}`);
   }
 }
 
 async function integration() {
+  buildNewServer();
+  htmlTransform = require('../server/new-server/transforms/dist/transform')
```

Comment:
Where does this come from? Is it something that `buildNewServer()` builds?

Reply:
`buildNewServer()` generates this file from the `TypeScript` transforms.

Reply:
@erwinmombay I just saw this old review comment thread in relation to #32865. I think this should be moved to `server.js`. The integration test runner shouldn't have to worry about server startup logic.